src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown

```diff
@@ -0,0 +1,370 @@
+---
+layout: updates/post
+title: "Access USB devices on the Web"
+description: "The WebUSB API makes USB safer and easier to use by bringing it to the Web."
+featured_image: /web/updates/images/2016-03-02-access-usb-devices-on-the-web/web-usb-hero.jpg
+published_on: 2016-08-12
+updated_on: 2016-08-12
+authors:
+  - beaufortfrancois
+tags:
+  - news
+  - USB
+  - WebUSB
+  - IoT
+  - Arduino
+  - origintrials
+---
+
+If I said plain and simple "USB", there is a good chance that you will
+immediately think of keyboards, mice, audio, video and storage devices. You're
+right but you'll find other kinds of Universal Serial Bus (USB) devices out
+there.
+
+These non-standardized USB devices require hardware vendors to write native
+drivers and SDKs in order for you (the developer) to take advantage of them.
+Sadly this native code has historically prevented these devices from being used
+by the Web. And that's one of the reasons the WebUSB API has been created: to
+provide a way to expose USB device services to the Web. With this API, hardware
+manufacturers will be able to build cross-platform JavaScript SDKs for their
+devices.
+But most importantly this will **make USB safer and easier to use by bringing
+it to the Web**.
+
+Let's see what you could expect with the WebUSB API:
+
+1. Buy a USB device,
+2. Plug it into your computer,
+3. A notification appears right away, with the right website to go to for this device.
```

Comment:
The USB device provides a landing page URL and a set of allowed origins. Unfortunately the Android OS's permission model prevents us from reading the landing page URL from the device early enough to display it.